### PR TITLE
Update backstage latest

### PR DESCRIPTION
**Pull request by jackmaloney**
PR to bring HMCTS backstage inline with the latest version of backstage.

### CODE REVIEW

File: app-config.yaml [hunk 1]
```diff
@@ -45,18 +42,83 @@ backend:
...
-    #   apiBaseUrl: https://ghe.example.net/api/v3
-    #   token: ${GHE_TOKEN}
-
+# See README.md in the proxy-backend plugin for information on the configuration format
 proxy:
+  '/circleci/api':
+    target: https://circleci.com/api/v1.1
+    headers:
+      Circle-Token: ${CIRCLECI_AUTH_TOKEN}
+
```
Comment (timja) [inline app-config.yaml:65]:
```suggestion
```

File: app-config.yaml [hunk 2]
```diff
@@ -45,18 +42,83 @@ backend:
...
+    headers:
+      Authorization: ${JENKINS_BASIC_AUTH_HEADER}
+
+  '/travisci/api':
+    target: https://api.travis-ci.com
+    changeOrigin: true
+    headers:
+      Authorization: ${TRAVISCI_AUTH_TOKEN}
+      travis-api-version: '3'
+
```
Comment (timja) [inline app-config.yaml:77]:
```suggestion
```

File: app-config.yaml [hunk 3]
```diff
@@ -45,18 +42,83 @@ backend:
...
+    target: https://api.buildkite.com/v2/
+    headers:
+      Authorization: ${BUILDKITE_TOKEN}
+
+  '/sentry/api':
+    target: https://sentry.io/api/
+    allowedMethods: ['GET']
+    headers:
+      Authorization: ${SENTRY_TOKEN}
+
```
Comment (timja) [inline app-config.yaml:103]:
```suggestion
```

File: app-config.yaml [hunk 4]
```diff
@@ -45,18 +42,83 @@ backend:
...
+    target: https://your.airflow.instance.com/api/v1
+    headers:
+      Authorization: ${AIRFLOW_BASIC_AUTH_HEADER}
+
+  '/gocd':
+    target: https://your.gocd.instance.com/go/api
+    allowedMethods: ['GET']
+    allowedHeaders: ['Authorization']
+    headers:
+      Authorization: Basic ${GOCD_AUTH_CREDENTIALS}
```
Comment (timja) [inline app-config.yaml:121]:
```suggestion
```

File: app-config.yaml [hunk 5]
```diff
@@ -65,76 +127,235 @@ proxy:
 techdocs:
   builder: 'local' # Alternatives - 'external'
   generator:
-    runIn: 'local' # Alternatives - 'local'
+    runIn: 'docker'
```
Comment (timja) [inline app-config.yaml:130]:
```suggestion
    runIn: 'local'
```

File: app-config.yaml [hunk 6]
```diff
@@ -65,76 +127,235 @@ proxy:
...
+    type: 'local' # Alternatives - 'googleGcs' or 'awsS3' or 'azureBlobStorage' or 'openStackSwift'. Read documentation for using alternatives.
 
-jenkins:
-  instances:
-    - name: cft
-      baseUrl: http://localhost:8080
-      username: admin
-      apiKey: ${JENKINS_BASIC_AUTH_HEADER}
+sentry:
+  organization: HMCTS
```
Comment (timja) [inline app-config.yaml:137]:
```suggestion
```

File: app-config.yaml [hunk 7]
```diff
@@ -65,76 +127,235 @@ proxy:
...
-        clientSecret: ${AUTH_GITHUB_CLIENT_SECRET}
-    microsoft:
-      development:
-        clientId: ${AUTH_MICROSOFT_CLIENT_ID}
-        clientSecret: ${AUTH_MICROSOFT_CLIENT_SECRET}
-        tenantId: ${AUTH_MICROSOFT_TENANT_ID}
+rollbar:
+  organization: HMCTS
+  # NOTE: The rollbar-backend & accountToken key may be deprecated in the future (replaced by a proxy config)
+  accountToken: my-rollbar-account-token
```
Comment (timja) [inline app-config.yaml:142]:
```suggestion
```

File: app-config.yaml [hunk 8]
```diff
@@ -65,76 +127,235 @@ proxy:
...
+lighthouse:
+  baseUrl: http://localhost:3003
+
+kubernetes:
+  serviceLocatorMethod:
+    type: 'multiTenant'
+  clusterLocatorMethods:
+    - type: 'config'
+      clusters: []
+
```
Comment (timja) [inline app-config.yaml:153]:
```suggestion
```

File: app-config.yaml [hunk 9]
```diff
@@ -65,76 +127,235 @@ proxy:
...
+kafka:
```
Comment (timja) [inline app-config.yaml:154]: I suspect this isn't needed

File: app-config.yaml [hunk 10]
```diff
@@ -65,76 +127,235 @@ proxy:
...
+  clientId: backstage
+  clusters:
+    - name: cluster
+      brokers:
+        - localhost:9092
+
+allure:
+  baseUrl: http://localhost:5050/allure-docker-service
```
Comment (timja) [inline app-config.yaml:162]:
```suggestion
```

File: app-config.yaml [hunk 11]
```diff
@@ -65,76 +127,235 @@ proxy:
...
+  azure:
+    - host: dev.azure.com
+      token: ${AZURE_TOKEN}
+  #  googleGcs:
+  #    clientEmail: 'example@example.com'
+  #    privateKey: ${GCS_PRIVATE_KEY}
+  awsS3:
+    - host: amazonaws.com
+      accessKeyId: ${AWS_ACCESS_KEY_ID}
+      secretAccessKey: ${AWS_SECRET_ACCESS_KEY}
```
Comment (timja) [inline app-config.yaml:197]:
```suggestion
```

File: app-config.yaml [hunk 12]
```diff
@@ -65,76 +127,235 @@ proxy:
...
+    #     users:
+    #       dn: ou=people,ou=example,dc=example,dc=net
+    #       options:
+    #         filter: (uid=*)
+    #       map:
+    #         description: l
+    #     groups:
+    #       dn: ou=access,ou=groups,ou=example,dc=example,dc=net
+    #       options:
+    #         filter: (&(objectClass=some-group-class)(!(groupType=email)))
```
Comment (timja) [inline app-config.yaml:232]:
```suggestion
```

File: app-config.yaml [hunk 13]
```diff
@@ -65,76 +127,235 @@ proxy:
...
     microsoftGraphOrg:
+    ### Example for how to add your Microsoft Graph tenant
```
Comment (timja) [inline app-config.yaml:234]:
```suggestion
```

File: app-config.yaml [hunk 14]
```diff
@@ -65,76 +127,235 @@ proxy:
...
+azureDevOps:
+  host: dev.azure.com
+  token: my-token
+  organization: HMCTS
+
+apacheAirflow:
+  baseUrl: https://your.airflow.instance.com
+
+gocd:
+  baseUrl: https://your.gocd.instance.com
```
Comment (timja) [inline app-config.yaml:361]:
```suggestion
```

File: app-config.yaml [hunk 15]
```diff
@@ -65,76 +127,235 @@ proxy:
...
+      timezone: UTC
+    - label: NYC
+      timezone: 'America/New_York'
+    - label: STO
+      timezone: 'Europe/Stockholm'
+    - label: TYO
+      timezone: 'Asia/Tokyo'
+pagerduty:
+  eventsBaseUrl: 'https://events.pagerduty.com/v2'
+jenkins:
```
Comment (timja) [inline app-config.yaml:345]: probably not right but i'll check this when I test on Monday

File: app-config.yaml [hunk 16]
```diff
@@ -65,76 +127,235 @@ proxy:
...
+    - label: UTC
+      timezone: UTC
+    - label: NYC
+      timezone: 'America/New_York'
+    - label: STO
+      timezone: 'Europe/Stockholm'
+    - label: TYO
+      timezone: 'Asia/Tokyo'
+pagerduty:
+  eventsBaseUrl: 'https://events.pagerduty.com/v2'
```
Comment (timja) [inline app-config.yaml:344]:
```suggestion
```

File: app-config.yaml [hunk 17]
```diff
@@ -65,76 +127,235 @@ proxy:
...
+      username: backstage-bot
+      apiKey: 123456789abcdef0123456789abcedf012
+
+azureDevOps:
+  host: dev.azure.com
+  token: my-token
+  organization: HMCTS
+
+apacheAirflow:
+  baseUrl: https://your.airflow.instance.com
```
Comment (timja) [inline app-config.yaml:358]:
```suggestion
```

**Review by timja — COMMENTED** [review]
Lot's of default config we don't want

File: catalog-info.yaml [hunk 18]
```diff
@@ -1,7 +1,7 @@
 apiVersion: backstage.io/v1alpha1
 kind: Component
 metadata:
-  name: hmcts-backstage
+  name: backstage
```
Comment (timja) [inline catalog-info.yaml:4]: I think
```suggestion
  name: hmcts-backstage
```

File: packages/app/src/App.tsx [hunk 19]
```diff
@@ -69,16 +57,14 @@ const AppRouter = app.getRouter();
 const routes = (
   <FlatRoutes>
     <Navigate key="/" to="catalog" />
-    <Route path="/catalog" element={<CatalogIndexPage initiallySelectedFilter="all" />} />
+    <Route path="/catalog" element={<CatalogIndexPage />} />
```
Comment (timja) [inline packages/app/src/App.tsx:60]: can you re-add the filter it was there intentionally

```suggestion
    <Route path="/catalog" element={<CatalogIndexPage initiallySelectedFilter="all" />} />
```

**Review by timja — COMMENTED** [review]
haven't gone through everything

File: app-config.yaml [hunk 20]
```diff
@@ -65,76 +71,146 @@ proxy:
...
-          # Optional filter for users, use group membership to get users.
-          # This and userFilter are mutually exclusive, only one can be specified
-          #userGroupMemberFilter: "displayName eq 'Backstage Users'"
-          # Optional filter for group, see Microsoft Graph API for the syntax
-          # See https://docs.microsoft.com/en-us/graph/api/resources/group?view=graph-rest-1.0#properties
-          groupFilter: securityEnabled eq true
+          tenantId: ${MICROSOFT_GRAPH_TENANT_ID}
+          clientId: ${MICROSOFT_GRAPH_CLIENT_ID}
+          clientSecret: ${MICROSOFT_GRAPH_CLIENT_SECRET_TOKEN}
+          userFilter: accountEnabled eq true and userType eq 'member'
```
Comment (timja) [inline app-config.yaml:109]: can you revert the userFilter and groupFilter changes please?

justice users are guests so we can't filter on member only

Comment (jackmaloney) [reply app-config.yaml]: Done

File: packages/app/package.json [hunk 21]
```diff
@@ -4,38 +4,55 @@
...
+    "@backstage/plugin-catalog-import": "^0.8.4",
+    "@backstage/plugin-catalog-react": "^0.8.0",
+    "@backstage/plugin-circleci": "^0.3.1",
+    "@backstage/plugin-cloudbuild": "^0.3.1",
+    "@backstage/plugin-code-coverage": "^0.1.28",
+    "@backstage/plugin-github-actions": "^0.5.1",
+    "@backstage/plugin-jenkins": "^0.7.0",
+    "@backstage/plugin-kubernetes": "^0.6.1",
+    "@backstage/plugin-lighthouse": "^0.3.1",
+    "@backstage/plugin-newrelic-dashboard": "^0.1.9",
```
Comment (timja) [inline packages/app/package.json:29]: can we remove some of the plugins here we don't need, things like circle ci, cloudbuild, azuredevops, newrelic etc?

File: packages/app/package.json [hunk 22]
```diff
@@ -4,38 +4,55 @@
...
+    "@backstage/plugin-techdocs": "^0.15.0",
+    "@backstage/plugin-todo": "^0.2.3",
+    "@backstage/plugin-user-settings": "^0.4.0",
+    "@backstage/theme": "^0.2.15",
     "@material-ui/core": "^4.12.2",
     "@material-ui/icons": "^4.9.1",
-    "@roadiehq/backstage-plugin-github-pull-requests": "^1.1.2",
+    "@roadiehq/backstage-plugin-github-insights": "^1.5.0",
+    "@roadiehq/backstage-plugin-github-pull-requests": "^1.4.0",
+    "@roadiehq/backstage-plugin-travis-ci": "^1.3.6",
```
Comment (timja) [inline packages/app/package.json:45]: remove this and next?

**Review by timja — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Pull request merged**